Common/AM_datatypes.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+struct AM_SM_Commands {
+};
+
+struct SM_AM_Commands {
+    uint8_t motor_outputs[12];
+};
+
+struct PIDValues {
+    double P;
+    double I;
+    double D;
+};
```

Comment:
if we're making this generic you might want to add in an "A" for advanced. Unlikely that we'll ever need to use it but if you don't want to have to change stuff later.